pages/api/trips/[id].js

```diff
@@ -0,0 +1,41 @@
+import dbConnect from "../../../lib/dbConnect";
+import Trip from "../../../models/Trip";
+import { getTripById } from "../../../services/tripService";
+
+export default async function handler(req, res) {
+  const {
+    query: { id },
+    method,
+  } = req;
+
+  await dbConnect();
+
+  if (!id.match("^[0-9a-fA-F]{24}$") || !id.match("^[a-z0-9]*$"))
+    return res.status(400).json({ error: "Wrong id" });
+
```

Comment:
Why do you check the format of the id?

Reply:
It's a validation to protect the API

Reply:
I reduced the regEx. The first condition is sufficient